pkg/login/social/generic_oauth.go

```diff
@@ -160,6 +166,14 @@ func (s *SocialGenericOAuth) UserInfo(client *http.Client, token *oauth2.Token)
 			s.log.Debug("Setting user info groups from extracted groups")
 			userInfo.Groups = groups
 		}
+
+		teams, err := s.extractTeams(data)
+		if err != nil {
+			s.log.Error("Failed to extract teams", "error", err)
+		} else if len(teams) > 0 {
+			s.log.Debug("Setting user info teams from extracted teams")
+			userInfo.Teams = teams
```

Comment:
Suggested moving this to a separate PR in another comment, but to me it seems like you can use a variable for teams rather than putting it into `BasicUserInfo`.